core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/DataPoint.java

```diff
@@ -1,179 +1,218 @@
     {
         this.InitializeFields();
     }
-
+    
+    /**
+     * Gets the Name property.
+     */
     public String getName() {
         return this.name;
     }
-
+    
+    /**
+     * Sets the Name property.
+     */
     public void setName(String value) {
         this.name = value;
     }
-
+    
+    /**
+     * Gets the Kind property.
+     */
     public DataPointType getKind() {
         return this.kind;
     }
-
+    
+    /**
+     * Sets the Kind property.
+     */
     public void setKind(DataPointType value) {
         this.kind = value;
     }
-
+    
+    /**
+     * Gets the Value property.
+     */
     public double getValue() {
         return this.value;
     }
-
+    
+    /**
+     * Sets the Value property.
+     */
     public void setValue(double value) {
         this.value = value;
     }
-
+    
+    /**
+     * Gets the Count property.
+     */
     public Integer getCount() {
         return this.count;
     }
-
+    
+    /**
+     * Sets the Count property.
+     */
     public void setCount(Integer value) {
         this.count = value;
     }
-
+    
+    /**
+     * Gets the Min property.
+     */
     public Double getMin() {
         return this.min;
     }
-
+    
+    /**
+     * Sets the Min property.
+     */
     public void setMin(Double value) {
         this.min = value;
     }
-
+    
+    /**
+     * Gets the Max property.
+     */
     public Double getMax() {
         return this.max;
     }
-
+    
+    /**
+     * Sets the Max property.
+     */
     public void setMax(Double value) {
         this.max = value;
     }
-
+    
+    /**
+     * Gets the StdDev property.
+     */
     public Double getStdDev() {
         return this.stdDev;
     }
-
+    
+    /**
+     * Sets the StdDev property.
+     */
     public void setStdDev(Double value) {
         this.stdDev = value;
     }
+    
 
     /**
      * Serializes the beginning of this object to the passed in writer.
      * @param writer The writer to serialize this object to.
      */
     @Override
-    public void serialize(JsonTelemetryDataSerializer writer) throws IOException {
-        Preconditions.checkNotNull(writer, "writer should be a nno-null value");
-
-        serializeContent(writer);
+    public void serialize(JsonTelemetryDataSerializer writer) throws IOException
+    {
+        Preconditions.checkNotNull(writer, "writer must be a non-null value");
+        this.serializeContent(writer);
     }
 
     /**
      * Serializes the beginning of this object to the passed in writer.
      * @param writer The writer to serialize this object to.
-     * @throws IOException Might be throw during serialization.
      */
-    protected void serializeContent(JsonTelemetryDataSerializer writer) throws IOException {
+    protected void serializeContent(JsonTelemetryDataSerializer writer) throws IOException
+    {
         writer.write("name", name);
-
-        if (!DataPointType.Measurement.equals(kind)) {
```

Comment:
I think it was an optimization to not send the value that has a default: https://github.com/Microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/DataPoint.bond#L13 Shouldn't break anything, I'll thing how to improve code generator